packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -36,284 +37,119 @@ const
-			 *
-			 * @type {Object}
-			 * @param {Object} ref
-			 * @private
-			 */
-			initUiChildRef: PropTypes.func,
-
-			/**
-			 * Prop to check if horizontal Scrollbar exists or not.
-			 *
-			 * @type {Boolean}
-			 * @private
-			 */
-			isHorizontalScrollbarVisible: PropTypes.bool,
-
-			/**
-			 * Prop to check if vertical Scrollbar exists or not.
-			 *
-			 * @type {Boolean}
-			 * @private
-			 */
-			isVerticalScrollbarVisible: PropTypes.bool,
-
-			/**
-			 * The array for individually sized items.
-			 *
-			 * @type {Number[]}
-			 * @private
-			 */
-			itemSizes: PropTypes.array,
-
-			/**
-			 * It scrolls by page when `true`, by item when `false`.
-			 *
-			 * @type {Boolean}
-			 * @default false
-			 * @private
-			 */
-			pageScroll: PropTypes.bool,
-
-			/**
-			 * The ARIA role for the list.
-			 *
-			 * @type {String}
-			 * @default 'list'
-			 * @public
-			 */
-			role: PropTypes.string,
-
-			/**
-			 * `true` if rtl, `false` if ltr.
-			 * Normally, [Scrollable]{@link ui/Scrollable.Scrollable} should set this value.
-			 *
-			 * @type {Boolean}
-			 * @private
-			 */
-			rtl: PropTypes.bool,
-
-			/**
-			 * Spacing between items.
-			 *
-			 * @type {Number}
-			 * @default 0
-			 * @public
-			 */
-			spacing: PropTypes.number,
-
-			/**
-			 * Spotlight Id. It would be the same with [Scrollable]{@link ui/Scrollable.Scrollable}'s.
-			 *
-			 * @type {String}
-			 * @private
-			 */
-			spotlightId: PropTypes.string,
-
-			/**
-			 * When it's `true` and the spotlight focus cannot move to the given direction anymore by 5-way keys,
-			 * a list is scrolled with an animation to the other side and the spotlight focus moves in wraparound manner.
-			 *
-			 * When it's `'noAnimation'`, the spotlight focus moves in wraparound manner as same as when it's `true`
-			 * except that a list is scrolled without an animation.
-			 *
-			 * @type {Boolean|String}
-			 * @default false
-			 * @public
-			 */
-			wrap: PropTypes.oneOfType([
-				PropTypes.bool,
-				PropTypes.oneOf(['noAnimation'])
-			])
-		}
-
-		static defaultProps = {
-			dataSize: 0,
-			focusableScrollbar: false,
-			pageScroll: false,
-			spacing: 0,
-			wrap: false
+		// Instance variables
+		const variables = useRef({
+			isScrolledBy5way: false,
+			isScrolledByJump:  false,
+			isWrappedBy5way:  false,
+			lastFocusedIndex:  null,
+			nodeIndexToBeFocused:  null,
+			preservedIndex:  null,
+			restoreLastFocused:  false,
+			uiRefCurrent:  null,
+			preventScroll:  null,
+			pause: null
+		});
+
+		if (variables.current.pause === null) {
+			variables.current.pause = new Pause('VirtualListBase');
 		}
 
-		constructor (props) {
-			super(props);
-
-			const {spotlightId} = props;
-			if (spotlightId) {
-				this.configureSpotlight(spotlightId);
-			}
-
-			this.pause = new Pause('VirtualListBase');
+		// Constructor
+		const {spotlightId} = props;
+		if (spotlightId) {
+			configureSpotlight();
```

Comment:
I believe this can be skipped since this method will be called in useEffect later anyway.